bat-utils/lib/runtime-database.js

```diff
@@ -128,27 +128,51 @@ Database.prototype.get = function (collection, debug) {
 Database.prototype.checkIndices = async function (debug, entries) {
   entries.forEach(async (entry) => {
     const category = entry.category
-    let doneP, indices
+    let doneP, indices, status
 
-    try { indices = await category.indexes() } catch (ex) { indices = [] }
-    doneP = underscore.keys(indices).indexOf(entry.property + '_1') !== -1
+    const form = (index) => {
+      let result = ''
 
-    debug(entry.name + ' indices ' + (doneP ? 'already' : 'being') + ' created')
+      underscore.keys(index).forEach((key) => { result += '_' + key + '_' + index[key] })
+      return result.substr(1)
+    }
+
+    const gather = (list) => {
```

Comment:
why do `form` and `gather` need to be defined inside every entries.forEach call? can you hoist them outside the scope into utility functions?

Reply:
good catch! they don't. i will move them up one scope.